src/main/java/com/catapi/service/TranslationService.java

```diff
@@ -0,0 +1,124 @@
+package com.catapi.service;
+
+import com.catapi.entity.CatFact;
+import com.catapi.entity.CatFactTranslation;
+import com.catapi.enums.ActiveState;
+import com.catapi.enums.Locale;
+import com.catapi.enums.UpdateMode;
+import com.catapi.exception.ExternalApiException;
+import com.catapi.exception.TranslationException;
+import com.catapi.jpa.CatFactRepository;
+import com.catapi.jpa.CatFactTranslationRepository;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.*;
+import org.springframework.stereotype.Service;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Slf4j
+@Service
+public class TranslationService {
+
+    public static final String LINGUATOOLS_URL = "https://linguatools.net/translators/translation_startseite";
+    private final CatFactTranslationRepository catFactTranslationRepository;
+    private final CatFactRepository catFactRepository;
+    private final OkHttpClient okHttpClient;
+
+    public TranslationService(
+            CatFactTranslationRepository catFactTranslationRepository,
+            CatFactRepository catFactRepository, OkHttpClient okHttpClient) {
+        this.catFactTranslationRepository = catFactTranslationRepository;
+        this.catFactRepository = catFactRepository;
+        this.okHttpClient = okHttpClient;
+    }
+
+    public String translate(Locale locale, String text) {
+        return getLinguatoolsTranslation(locale, text);
+    }
+
+    public void translateAllCatFactsByLinguatools(Locale locale) {
+        final List<CatFact> allFacts = catFactRepository.findAllByActiveState(ActiveState.ACTIVE);
+        allFacts.forEach(fact -> {
+            Optional<CatFactTranslation> localeTranslation = fact.getCatFactTranslations().stream()
+                    .filter(catFactTranslation -> catFactTranslation.getLocale() == locale)
+                    .findFirst();
+
+            localeTranslation.ifPresentOrElse(
+                    _ -> log.debug("Translation is already present. No need to translate"),
+                    () -> {
+                        String textToTranslate = fact.getFact();
+                        String translationText = getLinguatoolsTranslation(locale, textToTranslate);
+                        createCatFactTranslation(fact, locale, translationText);
+                    });
+        });
+        log.info(STR."All active facts without translations to \{locale.getLanguageName()} are translated");
+    }
```

Comment:
This function tries to find every active cat fact, then looks for all of translations and if there is no translation with such locale (specified in the method argument), it translates it and save in DB.

So it creates new translation only in case when it did not exist before